addons/base_automation/models/base_automation.py

```diff
@@ -393,6 +425,91 @@ def unlink(self):
         self._update_registry()
         return res
 
+    def action_rotate_webhook_uuid(self):
+        for automation in self:
+            automation.webhook_uuid = str(uuid4())
+
+    def action_view_webhook_logs(self):
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_window',
+            'name': _('Webhook Logs'),
+            'res_model': 'ir.logging',
+            'view_mode': 'tree,form',
+            'domain': [('path', '=', "base_automation(%s)" % self.id)],
+        }
+
+    def _execute_webhook(self, payload):
+        """ Execute the webhook for the given payload.
+        The payload is a dictionnary that can be used by the `record_getter` to
+        identify the record on which the automation should be run.
+        """
+        self.ensure_one()
+        ir_logging_sudo = self.env['ir.logging'].sudo()
+
+        # info logging is done by the ir.http logger
+        msg = "Webhook #%s triggered with payload %s"
+        msg_args = (self.id, payload)
+        _logger.debug(msg, *msg_args)
+        if self.log_webhook_calls:
+            ir_logging_sudo.create({'name': _("Webhook Log"),
+                        'type': 'server',
+                        'dbname': self._cr.dbname,
+                        'level': 'INFO',
+                        'message': msg % msg_args,
+                        'path': "base_automation(%s)" % self.id,
+                        'func': '',
+                        'line': ''})
+        if self.record_getter:
+            try:
+                record = safe_eval.safe_eval(self.record_getter, self._get_eval_context(payload=payload))
+            except Exception as e: # noqa: BLE001
+                msg = "Webhook #%s could not be triggered because the record_getter failed:\n%s"
+                msg_args = (self.id, traceback.format_exc())
+                _logger.warning(msg, *msg_args)
+                if self.log_webhook_calls:
+                    ir_logging_sudo.create({'name': _("Webhook Log"),
+                                'type': 'server',
+                                'dbname': self._cr.dbname,
+                                'level': 'ERROR',
+                                'message': msg % msg_args,
+                                'path': "base_automation(%s)" % self.id,
+                                'func': '',
+                                'line': ''})
+                raise e
+        else:
+            record = None
```

Comment:
move record = None to the top of the method, it should be the default value, that way you avoid the else